File: gui/frontend/src/lib/splash.test.ts
import { describe, it, expect } from "vitest";
import { centerInBox } from "./splash";

describe("centerInBox", () => {
  // Regression: the version was hard-truncated to 6 chars, so "v0.2.10" (7
  // chars, the first two-digit patch) rendered as "v0.2.1" on the splash.
  it("keeps the full version and fills the exact box width", () => {
    const r = centerInBox("v0.2.10", 64);
    expect(r.length).toBe(64);
    expect(r.trim()).toBe("v0.2.10");
  });

  it("centers the version within the field", () => {
    const r = centerInBox("v0.2.10", 64);
    const lead = r.length - r.trimStart().length;
    expect(lead).toBe(Math.floor((64 - "v0.2.10".length) / 2)); // 28
  });

  it("previous ≤6-char versions still fill the width", () => {
    for (const v of ["dev", "v0.2.9"]) {
      const r = centerInBox(v, 64);
      expect(r.length).toBe(64);
      expect(r.trim()).toBe(v);
    }
  });

  it("clips an absurdly long version so the border never overflows", () => {
    expect(centerInBox("x".repeat(100), 64).length).toBe(64);
  });
});

File: internal/ui/splash_test.go
package ui

import (
	"strings"
	"testing"
)

// TestCenterVersion guards the splash version field. Regression: "v0.2.10" (7
// chars, the first two-digit patch number) was truncated to "v0.2.1" by a fixed
// 6-char slot.
func TestCenterVersion(t *testing.T) {
	got := centerVersion("v0.2.10", splashBoxInterior)
	if len(got) != splashBoxInterior {
		t.Errorf("width = %d, want %d", len(got), splashBoxInterior)
	}
	if strings.TrimSpace(got) != "v0.2.10" {
		t.Errorf("trimmed = %q, want v0.2.10 (no truncation)", strings.TrimSpace(got))
	}

	// Previous ≤6-char versions still fill the field exactly.
	for _, v := range []string{"dev", "v0.2.9", ""} {
		g := centerVersion(v, splashBoxInterior)
		if len(g) != splashBoxInterior || strings.TrimSpace(g) != strings.TrimSpace(v) {
			t.Errorf("centerVersion(%q) = %q (len %d)", v, strings.TrimSpace(g), len(g))
		}
	}

	// An absurdly long version is clipped so the box border can't overflow.
	if l := len(centerVersion(strings.Repeat("x", 100), splashBoxInterior)); l != splashBoxInterior {
		t.Errorf("overlong width = %d, want %d", l, splashBoxInterior)
	}
}
